### PR TITLE
feat: SqlStatement uses `op<<(Value)`

### PR DESCRIPTION
**Pull request by devjgm**
Nobody should use `PrintTo(Value, os)`. This PR documents that users
should not call `PrintTo` and that it will be removed in an upcoming
release. All streaming of `Value` objects should be done using
`operator<<`.

Note this PR changes the output format of `SqlStatement`, and it
documents that the output format is for human consumption only and may
change without notice in the future (same as is documented for Value).

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1361)

### CODE REVIEW

File: google/cloud/spanner/value.h [hunk 1]
```diff
@@ -305,20 +305,11 @@ class Value {
...
-   *   machine consumption as the output format may change without notice.
-   */
-  friend void PrintTo(Value const& v, std::ostream* os);
-
   /**
    * Outputs string representation of a given Value to the provided stream.
    *
    * @warning This is intended for debugging and human consumption only, not
-   *   machine consumption as the output format may change without notice.
+   *     machine consumption as the output format may change without notice.
```
Comment (devbww) [inline google/cloud/spanner/value.h]: Nit: Now that you're editing this, you might as well put a comma after "consumption".

File: google/cloud/spanner/sql_statement.h [hunk 2]
```diff
@@ -103,8 +103,15 @@ class SqlStatement {
     return !(a == b);
   }
 
- private:
+  /**
+   * Outputs a string representation of the given @p stmt to the given @p os.
+   *
+   * @warning This is intended for debugging and human consumption only, not
+   *     machine consumption as the output format may change without notice.
```
Comment (devbww) [inline google/cloud/spanner/sql_statement.h]: See the other comment about a comma after "consumption".

File: google/cloud/spanner/value.h (hunk 1 above)
Comment (devjgm) [reply google/cloud/spanner/value.h]: Done.

File: google/cloud/spanner/sql_statement.h (hunk 2 above)
Comment (devjgm) [reply google/cloud/spanner/sql_statement.h]: Done

**Pull request merged**